<a href="https://colab.research.google.com/github/glgunderson/AI-Perceptions/blob/main/ai_perceptions.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!wget -O AI-111.xlsx "https://raw.githubusercontent.com/glgunderson/AI-Perceptions/main/data/AI-111.xlsx"

--2026-03-19 06:00:55--  https://raw.githubusercontent.com/glgunderson/AI-Perceptions/main/data/AI-111.xlsx
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 28471 (28K) [application/octet-stream]
Saving to: ‘AI-111.xlsx’

AI-111.xlsx         100%[===================>]  27.80K  --.-KB/s    in 0.002s  

2026-03-19 06:00:55 (16.3 MB/s) - ‘AI-111.xlsx’ saved [28471/28471]



In [ ]:
import pandas as pd
df = pd.read_excel("AI-111.xlsx")
df.head()

,Age,Education,Frequency,Job status,Field,Field_10_TEXT,Perceptions_1,Perceptions_2,Perceptions_3,Perceptions_4,...,Procedural Fairness_2,Outcome Fairness_1,Outcome Fairness_2,Outcome Fairness_3,Performance Pressure_1,Performance Pressure_2,Performance Pressure_3,Competency_1,Competency_2,Competency_3
0,29,1,5,5,1,NaN,5,4,4,4,...,4,1,5,1,4,4,4,2,5,5
1,20,1,2,1,"7,8,9",NaN,4,3,3,3,...,4,3,3,2,4,2,3,2,2,3
2,20,1,4,5,"1,2,3,4,9",NaN,4,4,4,4,...,3,2,2,3,4,4,4,4,4,4
3,23,3,5,1,"1,4",NaN,4,4,4,5,...,3,2,4,3,4,4,4,4,4,5
4,20,1,3,5,"1,2,3,4,9",NaN,4,4,4,4,...,4,4,4,4,4,4,2,4,4,4


In [ ]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 49 columns):
 #   Column                                              Non-Null Count  Dtype 
---  ------                                              --------------  ----- 
 0   Age                                                 110 non-null    int64 
 1   Education                                           110 non-null    int64 
 2   Frequency                                           110 non-null    int64 
 3   Job status                                          110 non-null    int64 
 4   Field                                               110 non-null    object
 5   Field_10_TEXT                                       7 non-null      object
 6   Perceptions_1                                       110 non-null    int64 
 7   Perceptions_2                                       110 non-null    int64 
 8   Perceptions_3                                       110 non-null    int64 
 9   Perception

,Age,Education,Frequency,Job status,Perceptions_1,Perceptions_2,Perceptions_3,Perceptions_4,Perceptions_5,AI Expectations_1,...,Procedural Fairness_2,Outcome Fairness_1,Outcome Fairness_2,Outcome Fairness_3,Performance Pressure_1,Performance Pressure_2,Performance Pressure_3,Competency_1,Competency_2,Competency_3
count,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,...,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000,110.000000
mean,22.527273,1.372727,3.927273,2.536364,3.890909,3.754545,3.618182,4.172727,4.463636,4.081818,...,3.554545,2.572727,3.890909,2.454545,3.854545,3.872727,3.836364,3.909091,3.809091,3.881818
std,3.161434,0.647636,1.037897,1.769659,0.860397,0.869318,0.928467,0.822383,0.569521,0.791373,...,0.711458,1.035685,0.870995,1.001250,0.833214,0.847112,0.923965,0.883544,0.990741,0.786723
min,19.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,3.000000,2.000000,...,1.000000,1.000000,2.000000,1.000000,1.000000,2.000000,1.000000,1.000000,1.000000,1.000000
25%,21.000000,1.000000,3.000000,1.000000,4.000000,4.000000,3.000000,4.000000,4.000000,4.000000,...,3.000000,2.000000,3.250000,2.000000,3.250000,4.000000,3.000000,4.000000,3.000000,4.000000
50%,21.500000,1.000000,4.000000,2.000000,4.000000,4.000000,4.000000,4.000000,4.500000,4.000000,...,4.000000,2.000000,4.000000,2.000000,4.000000,4.000000,4.000000,4.000000,4.000000,4.000000
75%,23.000000,2.000000,5.000000,5.000000,4.000000,4.000000,4.000000,5.000000,5.000000,5.000000,...,4.000000,3.000000,4.000000,3.000000,4.000000,4.000000,4.000000,4.000000,4.000000,4.000000
max,37.000000,3.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,...,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000


In [ ]:
!pip install pingouin

In [ ]:
PP = df[['Performance Pressure_1', 'Performance Pressure_2', 'Performance Pressure_3']]
CE = df[['Competency_1', 'Competency_2', 'Competency_3']]

In [ ]:
import pingouin as pg

pg.cronbach_alpha(PP)
pg.cronbach_alpha(CE)

(np.float64(0.6895957913859663), array([0.574, 0.778]))

Cronbach’s alpha for the Performance Pressure (PP) scale was 0.69, indicating marginal internal consistency. While close to the commonly accepted threshold of 0.70, the relatively wide confidence interval suggests variability in item coherence.

In [ ]:
PP12 = df[['Performance Pressure_1', 'Performance Pressure_2']]
CE12 = df[['Competency_1', 'Competency_2']]

pg.cronbach_alpha(PP12)
pg.cronbach_alpha(CE12)

(np.float64(0.7119997561644671), array([0.58 , 0.803]))

The Performance Pressure scale demonstrated marginal internal consistency (α = 0.69). Item-level analysis indicated that the third item exhibited weaker alignment with the other indicators, suggesting some variation in how different dimensions of performance expectations are perceived.

*AI may increase different types of pressure, not one single dimension.*

In [ ]:
PP13 = df[['Performance Pressure_1', 'Performance Pressure_3']]
CE13 = df[['Competency_1', 'Competency_3']]


pg.cronbach_alpha(PP13)
pg.cronbach_alpha(CE13)

(np.float64(0.46052933351681125), array([0.213, 0.63 ]))

In [ ]:
PP23 = df[['Performance Pressure_2', 'Performance Pressure_3']]
CE23 = df[['Competency_2', 'Competency_3']]


pg.cronbach_alpha(PP23)
pg.cronbach_alpha(CE23)

(np.float64(0.5670549581839905), array([0.369, 0.703]))

In [ ]:
df['PP_score'] = PP.mean(axis=1)
df['CE_score'] = CE.mean(axis=1)

In [ ]:
df[['PP_score', 'CE_score']].describe()

,PP_score,CE_score
count,110.000000,110.000000
mean,3.854545,3.866667
std,0.745393,0.699789
min,1.666667,1.666667
25%,3.666667,3.666667
50%,4.000000,4.000000
75%,4.333333,4.333333
max,5.000000,5.000000


In [ ]:
df[['PP_score', 'CE_score']].corr()

,PP_score,CE_score
PP_score,1.00000,0.55266
CE_score,0.55266,1.00000


In [ ]:
IV5 = df[['Perceptions_1','Perceptions_2','Perceptions_3','Perceptions_4','Perceptions_5']]
df['IV5_score'] = IV5.mean(axis=1)

pg.cronbach_alpha(IV5)

(np.float64(0.5709304051046149), array([0.43 , 0.686]))

## Construct Refinement

In [ ]:
IV345 = df[['Perceptions_3','Perceptions_4','Perceptions_5']]
df['IV345_score'] = IV345.mean(axis=1)

pg.cronbach_alpha(IV345)

(np.float64(0.5959358808290156), array([0.445, 0.711]))

In [ ]:
IV = df[['Perceptions_4','Perceptions_5']]
df['IV_score'] = IV.mean(axis=1)

pg.cronbach_alpha(IV)

(np.float64(0.7276776246023335), array([0.603, 0.813]))

Perceived AI integration was operationalized using two items reflecting the increasing prevalence and future dominance of AI in the workplace. These items demonstrated acceptable internal consistency (α = 0.73).